src/platform_impl/linux/wayland/window/shim.rs

```diff
@@ -188,6 +210,48 @@ impl WindowHandle {
         }
     }
 
+    pub fn set_user_attention(&self, request_type: Option<UserAttentionType>) {
+        let xdg_activation = match self.xdg_activation.as_ref() {
+            None => return,
+            Some(xdg_activation) => xdg_activation,
+        };
+
+        //  Unsetting urgency is done by compositor and if the token is already is present we don't
+        //  need to create one more, so performing eaerly return.
```

Comment:
This comment is a bit messed up. I'd also recommend simplifying it a bit. Could just fix the typos though.

```suggestion
        // Urgency is only removed by the compositor and there's no need to raise urgency when it was already raised.
```